torch/_inductor/autotune_process.py

```diff
@@ -166,71 +168,179 @@ class BenchmarkRequest:
-            )
-
-        run = getattr(mod, self.kernel_name).run
-
         if DEBUG:
             load_elapse = time.time() - start_ts
             start_ts = time.time()
 
         # create args and out tensor
         if output_tensor is None:
             assert len(input_tensors) == 0
-            if isinstance(self.input_tensors, List):
-                input_tensors = tuple(x.to_tensor() for x in self.input_tensors)
-            if isinstance(self.input_tensors, TensorMeta):
-                input_tensors = tuple(self.input_tensors.to_tensor())
-            assert isinstance(self.output_tensor, TensorMeta)
-            output_tensor = self.output_tensor.to_tensor()
+            input_tensors = tuple(x.to_tensor() for x in self.input_tensor_meta)
+            output_tensor = self.output_tensor_meta.to_tensor()
 
         if DEBUG:
             create_tensor_elapse = time.time() - start_ts
             start_ts = time.time()
 
-        def worker() -> float:
-            return run(
-                *input_tensors,
-                output_tensor,
-                *self.extra_args,
-                grid=self.grid,
-                num_stages=self.num_stages,
-                num_warps=self.num_warps,
-            )
-
-        out = do_bench(worker)
+        out = do_bench_using_profiling(
+            self.make_run_fn(*input_tensors, output_tensor=output_tensor)
+        )
         torch.cuda.synchronize()  # shake out any CUDA errors
 
         if DEBUG:
             bench_elapse = time.time() - start_ts
             print(
-                f"InChidProcess {self.module_cache_key}: load {load_elapse}, "
-                + f"create tensor {create_tensor_elapse}, bench {bench_elapse}"
+                f"InChidProcess {str(self)}: load {load_elapse}, "
+                + f"create tensor {create_tensor_elapse}, bench {bench_elapse}, "
+                + f"collected time {out}"
             )
+        self.cleanup_run_fn()
         return out
 
 
+class TritonBenchmarkRequest(BenchmarkRequest):
+    def __init__(
+        self,
+        kernel_name: str,
+        input_tensor_meta: List[TensorMeta],
+        output_tensor_meta: TensorMeta,
+        extra_args: Dict[str, Any],
+        module_path: str,  # the path of the module defining the triton kernel
+        module_cache_key: str,
+        grid: List[int],
+        num_stages: int,
+        num_warps: int,
+    ):
+        super().__init__(kernel_name, input_tensor_meta, output_tensor_meta, extra_args)
+        self.module_path = module_path
+        self.module_cache_key = module_cache_key
+        self.grid = grid
+        self.num_stages = num_stages
+        self.num_warps = num_warps
+
+    def make_run_fn(
+        self, *input_tensors: torch.Tensor, output_tensor: torch.Tensor
+    ) -> Callable[[], None]:
+        mod = PyCodeCache.load_by_key_path(self.module_cache_key, self.module_path)
+        if DEBUG:
+            print(
```

Comment:
Better to use logging with a DEBUG level here.

Reply:
I guess, this approach is inherited from the existing code.

Reply:
Yes this is from original code. I refactored it a bit so that TritonBenchmarkRequest and CUDABenchmarkRequest can share code. I'm not sure whether log works correctly when benchmark is in a subprocess so don't want to just change it. Could be a follow up task.